### PR TITLE
[1275] Renames for post-desktop world

### PR DESCRIPTION
**Pull request by blrnw3**
https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/single-cell-data-portal/1275

#### Reviewers
**Functional:** 
@MDunitz 

**Readability:** 
@danieljhegeman 

## Changes
Re-organise `backend` module to reflect new world of hosted-only code:
* `backend/czi_hosted` -> `server`
* Moved tests to -> `server/tests/`
* Merged the `backend/common` directories into `server/common`
* Update imports, make commands etc. to reflect changes
* Run black

## Validation
* Ran unit tests
* Ran build commands
* Ran some other make commands

## Next steps
* Remove make commands for desktop
 * update/simplify remaining make commands
 * remove desktop references from github actions
 * remove compatibility tests
 * remove desktop references from documentation

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -19,9 +19,9 @@ venv/
 cellxgene/
```
Comment (MDunitz) [inline .gitignore:19]: change to explorer?

File: Makefile [hunk 2]
```diff
@@ -2,7 +2,7 @@ include common.mk
 
 BUILDDIR := build
 CLIENTBUILD := $(BUILDDIR)/client
-CZIHOSTEDBUILD := $(BUILDDIR)/backend/czi_hosted
+CZIHOSTEDBUILD := $(BUILDDIR)/server
```
Comment (MDunitz) [inline Makefile:5]: rename to SERVERBUILD?

File: server/__init__.py [hunk 3]
```diff
@@ -1,14 +1,14 @@
 import logging
 import sys
-from backend.common.utils.utils import import_plugins
+from server.common.utils.utils import import_plugins
 
 __version__ = "0.16.7"
```
Comment (MDunitz) [inline server/__init__.py:5]: Can we update this to pull from the current commit hash?

**Comment by blrnw3** [comment]
* Removed refs to 'hosted' and equivalents (from docs/scripts, not from code yet) 
* Deleted docs directory

File: Makefile [hunk 4]
```diff
@@ -77,45 +71,45 @@ smoke-test-annotations:
...
 
 .PHONY: lint
-lint: lint-czi-hosted-server lint-client
+lint: lint-server lint-client
 
-.PHONY: lint-czi-hosted-server
-lint-czi-hosted-server: fmt-py
-	flake8 backend/czi_hosted --per-file-ignores='backend/test/fixtures/czi_hosted_dataset_config_outline.py:F821 backend/test/fixtures/czi_hosted_server_config_outline.py:F821 backend/test/performance/scale_test_annotations.py:E501'
+.PHONY: lint-server
+lint-server-server: fmt-py
```
Comment (MDunitz) [inline Makefile:88]:
```suggestion
lint-server: fmt-py
```

File: Makefile [hunk 5]
```diff
@@ -77,45 +71,45 @@ smoke-test-annotations:
...
 .PHONY: lint-client
 lint-client:
 	cd client && $(MAKE) lint
 
 # CREATING DISTRIBUTION RELEASE
 
-.PHONY: pydist-czi-hosted
-pydist-czi-hosted: build-czi-hosted
+.PHONY: pydist-server
+pydist-server: build-server
```
Comment (MDunitz) [inline Makefile:98]: I think this can just be `pydist`

**Review by MDunitz — COMMENTED** [review]
We may want to keep (or move) `docs/gene_sets.md` and `docs/gene_sets_example.csv`. 
We may also want to keep the .gifs/images in the docs directory. I think @MaximilianLombardo was working on some tutorials?


edit 2:53 - discussed/resolved over slack

File: server/__init__.py (hunk 3 above)
Comment (blrnw3) [reply server/__init__.py:5]: Not sure what you mean. Will follow up

**Review by MDunitz — COMMENTED** [review]
Can you revert all of the formatting changes to files under `fbs/NetEncoding`? Those files are generated by the FlatBuffers compiler and shouldnt be modified

**Comment by blrnw3** [comment]
ok everything should be good now. I updated the fmt command to exclude that directory.
Made some minor changes to readme too, we'll likely want to make more changes later

**Review by MDunitz — APPROVED** [review]
lgtm

**Pull request merged**